guidance/library/_json.py

```diff
@@ -94,20 +108,26 @@ def _gen_json_number(lm):
 
 
 @guidance(stateless=True)
-def _gen_json_string(lm):
-    string_chars = select(
-        [
-            char_range("a", "z"),
-            char_range("A", "Z"),
-            char_range("0", "9"),
-            *[c for c in "-_' ,.!?/[]{}():;"],
-            "\\n",
-            "\\t",
-            "\\\\",
-        ],
-        recurse=True,
-    )
-    return lm + '"' + string_chars + '"'
+def _gen_json_string(
+    lm,
+    min_length: Union[int, None] = None,
+    max_length: Union[int, None] = None,
+    regex: Union[str, None] = None,
+):
+    lm += '"'
+    if regex is not None:
+        assert min_length is None and max_length is None
+        lm += gen(regex=regex)
+    elif min_length is not None and max_length is not None:
```

Comment:
What if only one is non-none? I think default semantics in JSON is min=0, max=inf

Reply:
That's TBD.